modules/aleph/includes/aleph.debt.inc

```diff
@@ -38,14 +36,33 @@ function aleph_debt_list() {
       'invoice_number' => NULL,
       'type' => $debt->getType(),
     );
-    $result[$id] = new DingProviderDebt($id, $data);
+    $result[$debt->getId()] = new DingProviderDebt($debt->getId(), $data);
   }
 
   return $result;
 }
 
-function aleph_debt_payment_received() {
-  return array();
+/**
+ * Mark debts as paid.
+ *
+ * @param float $amount
+ *   The total amount paid.
+ * @param array $debt_ids
+ *   IDs of the paid debts.
+ * @param mixed $order_id
+ *   Order ID of the payment transaction, to be recorded in the backend system.
+ */
+function aleph_debt_payment_received($amount, array $debt_ids = array(), $order_id = NULL) {
+  $patron = aleph_get_patron_by_account();
+  $patron_handler = new AlephPatronHandler(aleph_client());
+  $patron_handler->setPatron($patron);
+
+  try {
+    $patron_handler->addPayment($amount, $order_id);
+  }
+  catch (Exception $e) {
+    watchdog_exception('aleph', $e);
```

Comment:
Is that the only thing we want to do if registering a payment fails?

[Another implementation](https://github.com/ding2/ding2/blob/master/modules/fbs/includes/fbs.debt.inc#L169) seems to return a boolean value.

I see that we also get a boolean value from `$patron_handler->addPayment()` but that is also disregarded.

Reply:
Fixed.